src/tscore/Random.cc

```diff
@@ -0,0 +1,29 @@
+/** @file
+
+  Pseudorandom Number Generator
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#include "tscore/Random.h"
+
+namespace ts
+{
+thread_local std::mt19937_64 Random::_engine{std::random_device{}()};
```

Comment:
std::mt19937_64 has a constructor that sets. default seed ( https://en.cppreference.com/w/cpp/numeric/random/mersenne_twister_engine/mersenne_twister_engine ).  I don't think we can be confident that _engine will get initialize, because our threads are not created by std::thread objects.  cppreference says thread-local objects are initialized "initialized as part of thread launch" ( https://en.cppreference.com/w/cpp/language/initialization#Non-local_variables ).  Seem likely the initialization is done in the std::thread constructor.